### PR TITLE
Fix issue #16048 - Allow NULL as default bit value.

### LINKED ISSUE #16048: Error 500 field of type bit
Hi,

We are facing an error when trying to modify a field of type BIT. 
phpMyAdmin-5.0.2-1
php73-php-common-7.3.16-1 (FPM)
Linux  4.4.217-1 #1 SMP  2020 x86_64 x86_64 x86_64 GNU/Linux
CentOS release 6.10 (Final)
Server version: Apache/2.4.34 (Red Hat) (Server MPM:     event)

Trace:
 ```
PHP Fatal error:  Uncaught TypeError: htmlspecialchars_decode() expects parameter 1 to be string, null given in /usr/share/phpMyAdmin/libraries/classes/Util.php:2863
Stack trace:
#0 /usr/share/phpMyAdmin/libraries/classes/Util.php(2863): htmlspecialchars_decode(NULL, 3)
#1 /usr/share/phpMyAdmin/libraries/tbl_columns_definition_form.inc.php(339): PhpMyAdmin\Util::convertBitDefaultValue(NULL)
#2 /usr/share/phpMyAdmin/libraries/classes/Controllers/Table/StructureController.php(573): include('/usr/share/phpM...')
#3 /usr/share/phpMyAdmin/libraries/classes/Controllers/Table/StructureController.php(218): PhpMyAdmin\Controllers\Table\StructureController->displayHtmlForColumnChange(Array, 'tbl_structure.p...', Object(Symfony\Component\DependencyInjection\ContainerBuilder))
#4 /usr/share/phpMyAdmin/tbl_structure.php(68): PhpMyAdmin\Controllers\Table\StructureController->indexAction(Object(Symfony\Component\DependencyInjection\ContainerBuilder))
#5 {main}
  thrown in /usr/share/phpMyAdmin/libraries/classes/Util.php on line 2863
```

Regards,

### PR DESCRIPTION
**Pull request by peterkroon**
Allow NULL as default bit value.

### Description

Please describe your pull request.

Fixes #16048

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: libraries/classes/Util.php [hunk 1]
```diff
@@ -2860,6 +2860,7 @@ public static function printableBitValue(int $value, int $length): string
      */
     public static function convertBitDefaultValue($bit_default_value)
     {
+        if ($bit_default_value === NULL) $bit_default_value = "";
```
Comment (williamdes) [inline libraries/classes/Util.php:2863]:
```suggestion
        if ($bit_default_value === null) {
            $bit_default_value = '';
       }
```

**Review by williamdes — APPROVED** [review]

File: libraries/classes/Util.php [hunk 2]
```diff
@@ -2860,6 +2860,7 @@ public static function printableBitValue(int $value, int $length): string
...
         return preg_replace("/^b'(\d*)'?$/", '$1', htmlspecialchars_decode($bit_default_value, ENT_QUOTES), 1);
```
Comment (williamdes) [inline libraries/classes/Util.php:2864]:
```suggestion
        return preg_replace("/^b'(\d*)'?$/", '$1', htmlspecialchars_decode((string) $bit_default_value, ENT_QUOTES), 1);
```
Would be the best fix

Comment (peterkroon) [reply libraries/classes/Util.php:2864]: Yes :-)

**Review by williamdes — APPROVED** [review]

**Comment by williamdes** [comment]
I ammerging it ;)

**Comment by williamdes** [comment]
Thank you for your contribution @peterkroon !

**Pull request merged**